examples/n_n_pipeline.py

```diff
@@ -0,0 +1,48 @@
+"""
+Script for filtering a dataset for molecules with single 
+nitrogen-nitrogen bonds
+
+Usage:
+    python n_n_pipeline.py --smiles-database database.smi
+"""
+
+import argparse
+from openeye import oechem
+from dance.run import run_dance
+
+def main():
+    """Builds and runs a pipeline."""
+    
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--smiles-database",
+                        type=str,
+                        required=True,
+                        help=("Name of a SMILES file containing molecules "
+                              "to process in the pipeline."))
+    args = parser.parse_args()
+
+    def relevant_if_has_single_bonded_nitrogen(mol):
+        subs = oechem.OESubSearch("[#7:1]-[#7:2]")
```

Comment:
You can define `subs` outside the function because it does not depend on the molecule in any way. Right now, the exact same object is being re-constructed for every call of the function (i.e. for every molecule). To be fair, I'm not sure how expensive the construction is though.

Edit: Oh, I see you did it in the other file -- do it here too :blush: